pkg/endpoint/policy.go

```diff
@@ -294,23 +294,23 @@ func (e *Endpoint) updateAndOverrideEndpointOptions(opts option.OptionMap) (opts
 }
 
 // Called with e.mutex UNlocked
-func (e *Endpoint) regenerate(context *regenerationContext) (retErr error) {
+func (e *Endpoint) regenerate(ctx *regenerationContext) (retErr error) {
```

Comment:
Please consider putting purely cosmetic changes like this in their own commits, just to simplify review when there are some changes that are functional vs. others that are not functional. I agree that this is a nice cleanup though, `ctx` should be the standard var name for the context, mostly just to avoid the package name conflict in case that comes up.

Reply:
This was done out of necessity to be able to reference `context.Canceled` in `updateRegenerationStatistics()` and `regenerate()`, but I'll split this off into a separate commit.